src/GraphRequest.ts

```diff
@@ -302,6 +303,30 @@ export class GraphRequest {
 		}
 	}
 
+	/**
+	 * @private
+	 * Checks if the content-type is present in the _headers property. If not present, defaults the content-type to application/json
+	 * @param none
+	 * @returns nothing
+	 */
+	private setHeaderContentType(): void {
+		if (!this._headers) {
+			this.header("Content-Type", "application/json");
+			return;
+		}
+		let isContentTypePresent = false;
+		const headerKeys = Object.keys(this._headers);
+		for (const headerKey of headerKeys) {
+			if (headerKey.toLowerCase() === "content-type") {
+				isContentTypePresent = true;
```

Comment:
@jobala, another opportunity for early return is at line 321.

Without defining `isContentTypePresent`, we can return directly here if "content-type" is found. Then, reaching line 325 will mean that the loop has finished without finding "content-type" (otherwise we would have hit the early return case), so we can skip the if check there and blindly assign "application/json".

So semantically equivalent code starting from 317:
```
		const headerKeys = Object.keys(this._headers);
		for (const headerKey of headerKeys) {
			if (headerKey.toLowerCase() === "content-type") {
				return; // content-type found, no update needed
			}
		}
		// Default the content-type to application/json in case the content-type is not present in the header
		this.header("Content-Type", "application/json");
```

@nikithauc, please either change the function logic to what I have above if it makes sense to you or add a break statement after setting `isContentTypePresent`, so that we don't continue iterating over headers after finding the searched header.

Reply:
Yes. That's right. Will make this change

Reply:
Added the change.